### PR TITLE
[PDF] final testing and fixes

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
Here the fixes of the bugs, which I detected after complex testing on local.

### CODE REVIEW

File: lib/qae_form_builder/question.rb [hunk 1]
```diff
@@ -114,7 +114,12 @@ def escaped_title
     end
 
     def escaped_context(pdf=false)
-      content = pdf ? delegate_obj.pdf_context : delegate_obj.context
+      content = if pdf && delegate_obj.pdf_context.present?
+        delegate_obj.pdf_context
+      else
+        delegate_obj.context
+      end
```
Comment (matthewford) [inline lib/qae_form_builder/question.rb]: `end` at 121, 6 is not aligned with `if` at 117, 16

File: lib/qae_form_builder/question.rb [hunk 2]
```diff
@@ -114,7 +114,12 @@ def escaped_title
     end
 
     def escaped_context(pdf=false)
-      content = pdf ? delegate_obj.pdf_context : delegate_obj.context
+      content = if pdf && delegate_obj.pdf_context.present?
+        delegate_obj.pdf_context
+      else
```
Comment (matthewford) [inline lib/qae_form_builder/question.rb]: Align `else` with `if`.

File: lib/qae_form_builder/question.rb [hunk 3]
```diff
@@ -114,7 +114,12 @@ def escaped_title
     end
 
     def escaped_context(pdf=false)
-      content = pdf ? delegate_obj.pdf_context : delegate_obj.context
+      content = if pdf && delegate_obj.pdf_context.present?
+        delegate_obj.pdf_context
```
Comment (matthewford) [inline lib/qae_form_builder/question.rb]: Use 2 (not -8) spaces for indentation.

File: app/pdf_generators/qae_pdf_forms/general/draw_elements.rb [hunk 4]
```diff
@@ -152,6 +152,8 @@ def render_answer_by_display(title, display)
   end
 
   def render_standart_answer_block(title)
+    title = title.present? ? title : FormPdf::UNDEFINED_TITLE
```
Comment (dreamfall) [inline app/pdf_generators/qae_pdf_forms/general/draw_elements.rb]: note: can be shorted to `title = title.presence || FormPdf::UNDEFINED_TITLE`

File: app/pdf_generators/qae_pdf_forms/general/question_pointer.rb [hunk 5]
```diff
@@ -475,6 +475,7 @@ def sub_question_block(sub_question, sub_answer)
   end
 
   def sub_question_block_without_title(sub_answer)
+    sub_answer.present? ? sub_answer : FormPdf::UNDEFINED_TITLE
```
Comment (dreamfall) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: What does this line do?

Comment (RuslanHamidullin) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: so, subanswer can be blank.
If it's blank - we display "Not answered yet ..." message pulled from FormPdf::UNDEFINED_TITLE

Comment (dreamfall) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: you're not assigning any variable here ;)

Comment (RuslanHamidullin) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: we pass it as parametr)
```
sub_question_block_without_title(sub_answer)
```

Comment (RuslanHamidullin) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: in:
```
    form_pdf.indent 22.mm do
      if sub_answers.length > 1
        sub_answers_by_type
      else
        first_sub_question = sub_answers[0][1]
        sub_question_block_without_title(first_sub_question)
      end
    end
```

Comment (RuslanHamidullin) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: so, in example above:
```
sub_answers[0][1] also can be blank
```

Comment (RuslanHamidullin) [inline app/pdf_generators/qae_pdf_forms/general/question_pointer.rb]: oh my god, you right :baby: 
Need to correct. Gonna fix

**Comment by RuslanHamidullin** [comment]
:bomb:

**Pull request merged**